src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -868,8 +873,15 @@ class MergeTreeData : public IStorage
     using MatcherFn = std::function<bool(const DataPartPtr &)>;
     PartitionCommandsResultInfo freezePartitionsByMatcher(MatcherFn matcher, const StorageMetadataPtr & metadata_snapshot, const String & with_name, const Context & context);
 
+    // Partition helpers
     bool canReplacePartition(const DataPartPtr & src_part) const;
 
+    virtual void dropPartition(const ASTPtr & partition, bool detach, bool drop_part, const Context & context) = 0;
+    virtual PartitionCommandsResultInfo attachPartition(const ASTPtr & partition, const StorageMetadataPtr & metadata_snapshot, bool part, const Context & context) = 0;
+    virtual void replacePartitionFrom(const StoragePtr & source_table, const ASTPtr & partition, bool replace, const Context & context) = 0;
+    virtual void movePartitionToTable(const StoragePtr & dest_table, const ASTPtr & partition, const Context & context) = 0;
+    virtual void fetchPartition(const ASTPtr & partition, const StorageMetadataPtr & metadata_snapshot, const String & from, const Context & query_context);
```

Comment:
Maybe add a comment that this method doesn't make sense for non replicated tables.

Reply:
I've added a comment.